Objective-C/Tests/URLEndpointListenerTest.m

```
@@ -0,0 +1,162 @@
+
+API_AVAILABLE(macos(10.12), ios(10.0))
+@interface URLEndpointListenerTest : ReplicatorTest
+typedef CBLURLEndpointListenerConfiguration Config;
+typedef CBLURLEndpointListener Listener;
+
+@end
+
+@implementation URLEndpointListenerTest
+
+- (Listener*) listenAt: (uint16)port {
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: port];
+    config.disableTLS = YES;
+    Listener* list = [[Listener alloc] initWithConfig: config];
+    
+    // start listener
+    NSError* err = nil;
+    Assert([list startWithError: &err]);
+    AssertNil(err);
+    
+    return list;
+}
+
+- (void) testPort {
+    // initialize a listener
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: kPort];
+    config.disableTLS = YES;
+    Listener* list = [[Listener alloc] initWithConfig: config];
+    AssertEqual(list.port, 0);
+    
+    // start listener
+    NSError* err = nil;
+    Assert([list startWithError: &err]);
+    AssertNil(err);
+    AssertEqual(list.port, kPort);
+    
+    // stops
+    [list stop];
+    AssertEqual(list.port, 0);
+}
+
+- (void) testEmptyPort {
+    // initialize a listener
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: 0];
+    config.disableTLS = YES;
+    Listener* list = [[Listener alloc] initWithConfig: config];
+    AssertEqual(list.port, 0);
+    
+    // start listener
+    NSError* err = nil;
+    Assert([list startWithError: &err]);
+    AssertNil(err);
+    Assert(list.port != 0);
+    
+    // stops
+    [list stop];
+    AssertEqual(list.port, 0);
+}
+
+- (void) testBusyPort {
+    Listener* list1 = [self listenAt: kPort];
+    
+    // initialize a listener at same port
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: kPort];
+    config.disableTLS = YES;
+    Listener* list2 = [[Listener alloc] initWithConfig: config];
+    
+    // already in use when starting the second listener
+    [self ignoreException:^{
```

Why do we need to ignore the exception here?

---

may be, the exception thrown from lite core might not be caught in between? will check

---

already existing port will throw exception from liter core. so the problem lies in hitting the exception break point

---

will keep this as same. and merging, will fix this if any issue exists.